common/src/main/java/com/wynntils/wynn/model/SpellModel.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.ChatPacketReceivedEvent;
+import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.mc.objects.ChatType;
+import com.wynntils.wynn.event.SpellCastedEvent;
+import com.wynntils.wynn.objects.SpellType;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public final class SpellModel extends Model {
```

Comment:
I know I just asked you to make this a model... But, #909 just happened. I think does qualify as a Handler? Since we take MC stuff, chat packets, parse it for spell events. Then, models can listen to the events provided by handlers. That is the text-book case of handlers, right? @magicus

Reply:
And @DonkeyBlaster, sorry for all these trouble! You got unlucky, and decided to do this while magicus is working on one of the biggest refactors of Artemis.

Reply:
#914 does just this.

Reply:
Yes, correct. ActionBar is definitely a text-book Handler. It was just waaaay harder to convert than I could imagine. I've spend more hours than I care to admit on #914...